### PR TITLE
Add new configuration filename - .robotidy

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #406

### CODE REVIEW

File: docs/releasenotes/4.2.0.rst [hunk 1]
```diff
@@ -31,6 +31,12 @@ With following file structure::
 The ``test.robot`` file will use ``root/pyproject.toml`` configuration file and ``test2.robot`` will use
 ``root/source/nested/pyproject.toml``.
 
+Dotfile configuration filename (#506)
+-------------------------------------
+
+It is now possible to use new configuration file name besides ``pyproject.toml`` - ``.robotidy``. It uses the same
+toml syntax but allows to skip ``tool.rootidy`` section name.
```
Comment (mnojek) [inline docs/releasenotes/4.2.0.rst:38]:
```suggestion
toml syntax but allows to skip ``tool.robotidy`` section name.
```

File: docs/source/configuration/config_file.rst [hunk 2]
```diff
@@ -73,6 +73,16 @@ Source paths can be configured via ``src`` parameter. If the path does not exist
 If you don't provide source paths in the cli, they will be taken from the closest configuration file. In a setup with
 multiple configuration files, the source paths from other configurations than the closest will be ignored.
 
+``.robotidy`` file uses the same toml syntax as ``pyproject.toml`` file but allows to skip ``tool.robotodiy`` section:
```
Comment (mnojek) [inline docs/source/configuration/config_file.rst:76]:
```suggestion
``.robotidy`` file uses the same toml syntax as ``pyproject.toml`` file but allows to skip ``tool.robotidy`` section:
```

**Review by mnojek — APPROVED** [review]

File: docs/source/configuration/config_file.rst (hunk 2 above)
Comment (bhirsz) [reply docs/source/configuration/config_file.rst:76]: So many way to insert typo into this name :D

Comment (mnojek) [reply docs/source/configuration/config_file.rst]: Yup :D I had similar case at work with word "relevancy". It was written in three different ways in one file :D All incorrect :D

**Pull request merged**